lua/nvim-tree/explorer/filters.lua

```diff
@@ -223,4 +224,33 @@ function Filters:should_filter(path, fs_stat, status)
     or bookmark(self, path, fs_stat and fs_stat.type, status.bookmarks)
 end
 
+--- Check if the given path should be filtered, and provide the reason why it was
+---@param path string Absolute path
+---@param fs_stat uv.fs_stat.result|nil fs_stat of file
+---@param status table from prepare
+---@return FILTER_REASON
+function Filters:should_filter_as_reason(path, fs_stat, status)
```

Comment:
I reckon `function Filters:should_filter(path, fs_stat, status)` should go away and be completely replaced by this new function.

Reply:
We could. But there's still one place that uses the ``should_filter`` function that doesn't require a ``reason``.
It's located at ``search-node.lua`` file.

Reply:
Got it.